bpf/lib/policy.h

```diff
@@ -69,13 +69,7 @@ __policy_can_access(const void *map, struct __ctx_buff *ctx, __u32 local_id,
 # endif
 
 # if defined(ENABLE_ICMP_RULE)
-			/* Convert from unsigned char to unsigned short
-			 * considering byte order(little-endian).
-			 * In the little-endian case, for example, 2byte data "AB"
-			 * convert to "BA".
-			 * Therefore, the "icmp_type" should be shifted not just casting.
-			 */
-			key.dport = (__u16)(icmphdr.type << 8);
+			key.dport = bpf_u8_to_be16((__u16)icmphdr.type);
```

Comment:
I don't mean to bike shed this too much, but now that you macro'd away the call to `bpf_htons` might it make sense to place the (__u16) cast **in** that macro? The macro is semantically saying "we'll do the cast for you" IMO.

Reply:
Thank you for your review!
I fixed it :)